### PR TITLE
Form to script fix + proposed interface + test fix

### PR DESCRIPTION
**Pull request by chlupnoha**

### CODE REVIEW

File: s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java [hunk 1]
```diff
@@ -12,5 +13,10 @@
 
     Map<String, Model> form2Script(Model inputScript, Question form, String moduleType);
 
+    //this is possible however we do not know about changed files
+    void form2ScriptNew(Model scripts, Question form, String moduleType);
```
Comment (blcham) [inline s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java:17]: form2ScriptNew -> form2Script   (We don't need to keep old version)
scripts -> script

File: s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java [hunk 2]
```diff
@@ -12,5 +13,10 @@
...
+
+    void registerListener(ModelChangedListener listener);
```
Comment (blcham) [inline s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java:19]: Can't we just not have a listener at all? The method form2Script has a parameter of type Model. Whoever uses this method can register the listener before he calls the script on the model, right? So we don't need to put it into the Transformer interface !?

void form2ScriptNew(Model scripts, Question form, String moduleType);

File: s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java [hunk 3]
```diff
@@ -12,5 +13,10 @@
 
     Map<String, Model> form2Script(Model inputScript, Question form, String moduleType);
 
+    //this is possible however we do not know about changed files
```
Comment (blcham) [inline s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java:16]: do not understand this comment. We know about the changed files as we have mapping ontology iri --> file before calling this script

**Review by blcham — COMMENTED** [review]
See my comments of the implementation.

File: s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java (hunk 3 above)
Comment (chlupnoha) [reply s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java:16]: I did not realize Model contains iri... I thought there could be a problem with the imports

File: s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java (hunk 2 above)
Comment (chlupnoha) [reply s-pipes-forms/src/main/java/cz/cvut/spipes/transform/Transformer.java:19]: Yea but the current solution builds up a new Model so, if we register listener on `scripts` the listener is lost. The `registerListener` method is only for the current implementation. I was not able to reimplement `form2Script` method to adjust `scripts` reference.

**Pull request closed without merging**